scripts/data_prep/convert_text_to_mds.py

```diff
@@ -0,0 +1,329 @@
+    no_wrap: bool,
+    max_workers: int,
+    compression: str,
+    processes: int,
+    args_str: str,
+    reprocess: bool,
+):
+    done_file_name = 'done'
+    is_remote_output = is_remote_path(output_folder)
+
+    object_names = get_object_names(input_folder)
+
+    # Check if the text files in the bucket have already been processed.
+    if not reprocess and is_already_processed(output_folder, done_file_name,
+                                              args_str, object_names):
+        print(
+            f'Input folder {input_folder} is already processed at {output_folder}.'
+        )
+        return
+
+    # Use a temporary local directory if the output is remote and there are more than 1 processes
+    local_output_folder = tempfile.TemporaryDirectory(
+    ).name if is_remote_output else output_folder
+
+    if processes > 1:
+        # Download and convert the text files in parallel
+        args = get_task_args(object_names, local_output_folder, input_folder,
+                             processes, tokenizer_name, concat_tokens, eos_text,
+                             bos_text, no_wrap, compression, max_workers)
+        with Pool(processes=processes) as pool:
+            pool.starmap(download_and_convert, args)
+
+        # Merge the mds shards from each of the processes into a single folder
+        merge_shard_groups(local_output_folder)
+    else:
+        download_and_convert(object_names, local_output_folder, input_folder,
+                             tokenizer_name, concat_tokens, eos_text, bos_text,
+                             no_wrap, compression, max_workers)
+
+    # Write a done file with the args and object names
+    write_done_file(local_output_folder, done_file_name, args_str, object_names)
+
+    if is_remote_output:
+        # Upload the local output to the remote location
+        output_object_store = cast(
+            ObjectStore, maybe_create_object_store_from_uri(output_folder))
+        _, _, output_folder_prefix = parse_uri(output_folder)
+        pattern = os.path.join(local_output_folder, '*')
+        files_to_upload = sorted(glob(pattern))
+
+        # TODO: Use multi threading to upload files?
+        for local_path in files_to_upload:
+            assert not os.path.isdir(local_path)
+            remote_path = os.path.join(output_folder_prefix,
+                                       os.path.basename(local_path))
+            output_object_store.upload_object(remote_path, local_path)
+
+
+if __name__ == '__main__':
+    args = parse_args()
+    main(tokenizer_name=args.tokenizer,
+         output_folder=args.output_folder,
+         input_folder=args.input_folder,
+         concat_tokens=args.concat_tokens,
+         eos_text=args.eos_text,
+         bos_text=args.bos_text,
+         no_wrap=args.no_wrap,
+         max_workers=args.max_workers,
+         compression=args.compression,
+         processes=args.processes,
+         reprocess=args.reprocess,
+         args_str=str(args))
```

Comment:
do we want to include all the args here? e.g. changing the max workers shouldn't affect whether we need to reprocess, right?

Reply:
Good point, I suppose we need to include all the args except max_workers, reprocess?

Reply:
I think thats right.